### PR TITLE
Cs/support 12674 check permission on web user

### PR DESCRIPTION
**Pull request by Charl1996**
## Product Description
This is a fix that allows location-assigned admin (or any role that has the "access_all_locations" permission for that matter) web users to see the complete list of web users when downloading the web users, and not just the web users that are assigned to the location(s) the currently logged-in web user is assigned to. 

(See the relevant [support ticket](https://dimagi-dev.atlassian.net/browse/SUPPORT-12674))

## Technical Summary
The role's permissions of the web user is now also considered in addition to that web user's assigned location(s) when that respective web user tries to download users. 

## Feature Flag
This does not affect only a specific FF.

### Safety story
- Unit tests
- Tested on staging

### Automated test coverage
New unit tests cover the "count_web_users" part (which is enough to conclude that the correct users will be downloaded, since the same view is used and cleaned in both scenarios).

### QA Plan
- No formal QA plan, except me testing on staging myself.

### Rollback instructions
- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by AmitPhulera — APPROVED** [review]
Thanks for the fix and for adding the tests 🎉

File: corehq/apps/users/tests/test_views.py [hunk 1]
```diff
@@ -201,3 +209,79 @@ def test_delete_phone_number(self):
...
+    view = 'count_web_users'
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.domain = 'test'
+        cls.domain_obj = create_domain(cls.domain)
+
+        set_toggle(FILTERED_BULK_USER_DOWNLOAD.slug, cls.domain, True, namespace='domain')
```
Comment (orangejenny) [inline corehq/apps/users/tests/test_views.py:225]: FYI you can alternatively decorate the test with `@flag_enabled('FILTERED_BULK_USER_DOWNLOAD')`

File: corehq/apps/users/tests/test_views.py [hunk 2]
```diff
@@ -201,3 +209,79 @@ def test_delete_phone_number(self):
...
+class TestCountWebUsers(TestCase):
+
+    view = 'count_web_users'
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.domain = 'test'
+        cls.domain_obj = create_domain(cls.domain)
```
Comment (orangejenny) [inline corehq/apps/users/tests/test_views.py:223]: Does this domain object need to exist? (Creating & deleting domains is slow.)

**Review by orangejenny — APPROVED** [review]
Thank you for the speedy fix @Charl1996

File: corehq/apps/users/tests/test_views.py (hunk 2 above)
Comment (Charl1996) [reply corehq/apps/users/tests/test_views.py:223]: Seems like there's a few places the domain_obj is required. I guess I could patch all those places, but then I feel I could just as well create the domain object.

**Pull request merged**